README.md

```diff
@@ -2124,13 +2124,99 @@ the build will fail if test coverage is below 100%.
 
 You should be able to do the typical stuff:
 
+### 1. Cloning repository:
+Fork the repository and run this(Only for SSH):
+```
+git clone git@github.com:${your_github_username}/foobara.git
+```
+Create a new branch for you to push into(branch name is not limited to this format)
+```
+git checkout -b develop/${your_github_username}
+```
+Now navigate to project directory
+
 ```
-git clone git@github.com:foobara/foobara
 cd foobara
+```
+### 2. Installing bundler:
+Bundler is the package manager for ruby like npm and pip are for JS and python respectively
+
+To install bundler run:
+```
+sudo gem install bundler
+```
+Check the installation:
+```
+bundler -v
+```
+### 2. Installing mise (LINUX/WSL)
+Mise is a software, which helps us manage and install different versions of other tools, in our case ruby.
+To setup mise following the script given below:
+
+```
+sudo apt update -y && sudo apt install -y gpg sudo wget curl
+sudo install -dm 755 /etc/apt/keyrings
+wget -qO - https://mise.jdx.dev/gpg-key.pub | gpg --dearmor | sudo tee /etc/apt/keyrings/mise-archive-keyring.gpg 1> /dev/null
+echo "deb [signed-by=/etc/apt/keyrings/mise-archive-keyring.gpg arch=amd64] https://mise.jdx.dev/deb stable main" | sudo tee /etc/apt/sources.list.d/mise.list
+sudo apt update
+sudo apt install -y mise
+```
+
+Verify if it's installed or not by running:
+```
+mise -v
+```
+It should print out an ascii-art saying "mise-en-place"
+### 3. Install required ruby version
+In this project, we require >=3.40 version of ruby and we will be installing the same now
+
+```
+mise use -g ruby@3.4
+```
+Sometimes, if you are using **WSL** you might encounter some build failure errors. This is due to some missing packages like openssl. Simply run this:
+
+```
+sudo apt update
+sudo apt install -y build-essential \
+  libssl-dev zlib1g-dev libreadline-dev libyaml-dev libxml2-dev \
+  libxslt1-dev libffi-dev libgdbm-dev autoconf bison
```

Comment:
Seems like we could just move all of these up above with gpg and wget. Actually, we could add git and move it as the first instruction. I don't think the preceding apt update is really necessary (but is for the apt install mise one)